cpu/nrf52/include/periph_cpu.h

```diff
@@ -264,6 +264,34 @@ void spi_twi_irq_register_spi(NRF_SPIM_Type *bus,
 void spi_twi_irq_register_i2c(NRF_TWIM_Type *bus,
                               spi_twi_irq_cb_t cb, void *arg);
 
+/**
+ * @brief   Acquire the shared I2C/SPI peripheral in I2C mode
+ *
+ * @param   bus bus to acquire exclusive access on
+ */
+void nrf5x_i2c_acquire(NRF_TWIM_Type *bus);
```

Comment:
```suggestion
void nrf5x_i2c_acquire(NRF_TWIM_Type *bus, spi_twi_irq_cb_t cb, void *arg);
```

The shared peripherals have a single IRQ vector, which then needs to be dispatched to the correct ISR function. For this, the ISR will read `_irq[num]` and `_irq_arg[num]` for SPI/I2C peripheral number `num`. Just updating those on acquire may be sufficient.